README.md

```diff
@@ -2,258 +2,61 @@
 
-In order to complete this challenge, you will be making requests to `napi.busbud.com`, Busbud's production API.  
-For all requests, you MUST provide the following HTTP headers:
+## App Architecture
 
-Header | Value
---- | ---
-Accept | `application/vnd.busbud+json; version=2; profile=https://schema.busbud.com/v2/`
-X-Busbud-Token | The token provided in the challenge invitation email.
+I decided to use MVVM, with a clear folder structure of each layer, but with some exceptions:
 
-### Init results
+* **Models**: In this folder I decided to have all the plain objects to hold the info, there is no logic here, just structs with the mapped data needed.
+* **Views**: All UI related code is here, sorted by screens, so you'll find 3 folders, onboarding, booking (search) and results.
+* **ViewModel**: In this folder we can find the single ViewModel used in the results screen, that's because it is a single endpoint to fetch the departures.
+* **API**: In this folder I put my API layer, I also set it in a way that it can grow by adding more endpoints to it and the ApiManager has generics so it can be used to fetch any object that conforms to the Decodable protocol.
+* **Utils**: In this folder you will find any utility class like extensions and helpers to create data or format data as needed in the views.
+* **Support Files**: Here we have OS related files and assets folder.
 
-To get departure results, search is initialized via the following endpoint:
+## Cache Handling
 
-`https://napi.busbud.com/x-departures/:origin/:destination/:outbound_date`
+Some missing stuff here, for example the lag of the repository or data manager classes. That was intentional, because given the use case of this app, we don't want to have cached data, we always want to have a fresh data each time the user search for departures. So I don't use the data manager to decide where to fetch the info from, we just go to fetch from the API.
 
-PATH PARAMS  
+## UI Tweaks
 
-- `origin` : Origin's geohash
-- `destination` : Destination's geohash
-- `outbound_date` : Outbound departure date, ISO 8061 date
+Even tho I'm not a designer, I tried to apply as much effort as possible to the UI/UX, because a very good looking app helps to retain users, also its behavior. So I added some UI tweaks in the code, for example the `UIImage.blur()` function and so many others so can help to build a good looking app.
 
-QUERY PARAMS:
+## Third Party Libraries
+Although some companies doesn't like to use third-party libraries inside of their projects, there are some libraries that are needed to not *reinvent the wheel*. In this project I just use 4 libraries:
+* **Alamofire** to handle the API requests
```

Comment:
Probably unnecessary, you're only using it to create a single GET request, that's like 2 lines!